src/Components/TestExplorer.fs

```diff
@@ -34,411 +165,797 @@ type TestItem with
+        match tryGetTrxPath projectPath with
+        | None -> Array.empty
+        | Some trxPath ->
+            let selector = trxSelector trxPath
+            extractTestDefinitionsFromSelector selector
+
+    let extractTestResult (xpathSelector: XPath.XPathSelector) (executionId: string) : TrxTestResult =
+        // NOTE: The test result's `testName` isn't always the full name. Some libraries handle it differently
+        // Thus, it must be extracted from the test deff
+        let className =
+            xpathSelector.SelectString
+                $"/t:TestRun/t:TestDefinitions/t:UnitTest[t:Execution/@id='{executionId}']/t:TestMethod/@className"
+
+        let testName =
+            xpathSelector.SelectString
+                $"/t:TestRun/t:TestDefinitions/t:UnitTest[t:Execution/@id='{executionId}']/t:TestMethod/@name"
+
+        let outcome =
+            xpathSelector.SelectString $"/t:TestRun/t:Results/t:UnitTestResult[@executionId='{executionId}']/@outcome"
+
+        let errorInfoMessage =
+            xpathSelector.TrySelectString
+                $"/t:TestRun/t:Results/t:UnitTestResult[@executionId='{executionId}']/t:Output/t:ErrorInfo/t:Message"
+
+        let errorStackTrace =
+            xpathSelector.TrySelectString
+                $"/t:TestRun/t:Results/t:UnitTestResult[@executionId='{executionId}']/t:Output/t:ErrorInfo/t:StackTrace"
+
+        let timing =
+            let duration =
+                xpathSelector.SelectString
+                    $"/t:TestRun/t:Results/t:UnitTestResult[@executionId='{executionId}']/@duration"
+
+            let success, ts = TimeSpan.TryParse(duration)
+
+            if success then ts else TimeSpan.Zero
+
+        { ExecutionId = executionId
+          FullTestName = TestName.fromPathAndTestName className testName
+          Outcome = outcome
+          ErrorMessage = errorInfoMessage
+          ErrorStackTrace = errorStackTrace
+          Timing = timing }
+
+
+
+    let extractTrxResults (trxPath: string) =
+        let xpathSelector = trxSelector trxPath
+
+        let trxDefToTrxResult (trxDef: TrxTestDef) =
+            extractTestResult xpathSelector trxDef.ExecutionId
+
+        extractTestDefinitionsFromSelector xpathSelector |> Array.map trxDefToTrxResult
+
+    let inferHierarchy (testDefs: TrxTestDef array) : TestName.NameHierarchy<TrxTestDef> array =
+        testDefs
+        |> Array.map (fun td -> {| FullName = td.FullName; Data = td |})
+        |> TestName.inferHierarchy
+
+
+
 
 module DotnetTest =
-    let runProject (tc: TestController) (projectWithTests: ProjectWithTests) : JS.Promise<ProjectWithTestResults> =
-        logger.Debug("Nunit project", projectWithTests)
 
+    let internal dotnetTest (projectPath: string) (additionalArgs: string array) =
+        Process.exec
+            "dotnet"
+            (ResizeArray(
+                [| "test"
+                   projectPath
+                   "--logger:\"trx;LogFileName=Ionide.trx\""
```

Comment:
nit: consider writing this file somewhere inside Ionide's per-extension cache directory, which is part of the `ExtensionContext` passed to the `activate` call in fsharp.fs. Some docs on this directory are available [here](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext.storagePath). This will keep the user's directory clean. I might also suggest a subdirectory in this `storagePath` for test-related temp files, and another subdirectory in that for some kind of workspace identifier to prevent clobbering across multiple VScode instances?

Reply:
Thank you for the kind words!

I can certainly checkout these improvements to the result file location. Your suggestions make a lot of sense. The results location is mostly an artifact of the previous implementation. 
I'll also checkout moving code to FSAC, but it'll probably take a bit since I'll need to familiarize myself with FSAC.